pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -155,17 +211,24 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 		}
 	}
 
-	lastTaskRun, err := r.retrieveTaskRun(ctx, build, buildRun)
-	if err != nil {
-		updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
-		return reconcile.Result{}, handleError(fmt.Sprintf("Failed to list existing TaskRuns from BuildRun: %v", buildRun.Name), err, updateErr)
-	}
+	lastTaskRun := &v1beta1.TaskRun{}
 
-	if lastTaskRun != nil {
-		// TODO: Make this safer
-		if len(lastTaskRun.Status.Conditions) > 0 {
-			taskRunStatus := lastTaskRun.Status.Conditions[0].Status
+	// for existing TaskRuns update the BuildRun Status, if there is no TaskRun, then create one
+	if err = r.client.Get(ctx, types.NamespacedName{Name: request.Name, Namespace: request.Namespace}, lastTaskRun); err != nil && apierrors.IsNotFound(err) {
+		generatedTaskRun, err := r.createTaskRun(ctx, build, buildRun)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
 
+		ctxlog.Info(ctx, "creating TaskRun from BuildRun", namespace, request.Namespace, name, generatedTaskRun.Name, "BuildRun", buildRun.Name)
+		if err = r.client.Create(ctx, generatedTaskRun); err != nil {
+			updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
+			return reconcile.Result{}, handleError("Failed to create TaskRun if no TaskRun for that BuildRun exists", err, updateErr)
+		}
+	} else {
+		ctxlog.Info(ctx, "TaskRun already exists", namespace, request.Namespace, name, request.Name, "jodido", lastTaskRun.Name)
```

Comment:
`"jodido"`

Reply:
oh boy, dont even translate this. I knew it was somewhere but I could not find this. Thanks.